src/pybind/mgr/dashboard_v2/controllers/rbd.py

```diff
@@ -68,3 +95,35 @@ def get(self, pool_name):
         if status == ViewCache.VALUE_EXCEPTION:
             raise value
         return {'status': status, 'value': value}
+
+    def create(self, data):
+        if not self.rbd:
+            self.rbd = rbd.RBD()
+
+        # Get input values
+        name = data.get('name')
+        pool_name = data.get('pool_name')
+        size = data.get('size')
+        obj_size = data.get('obj_size')
+        features = data.get('features')
+        stripe_unit = data.get('stripe_unit')
+        stripe_count = data.get('stripe_count')
+        data_pool = data.get('data_pool')
+
+        # Set order
+        order = None
+        if obj_size and obj_size > 0:
+            order = int(round(math.log(float(obj_size), 2)))
+
+        # Set features
+        feature_bitmask = self._format_features(features)
+
+        ioctx = mgr.rados.open_ioctx(pool_name)
+
+        try:
+            self.rbd.create(ioctx, name, size, order=order, old_format=False,
+                            features=feature_bitmask, stripe_unit=stripe_unit,
+                            stripe_count=stripe_count, data_pool=data_pool)
+            return {'success': True, 'detail': 'Image {!s} created successfully'.format(name)}
```

Comment:
Why do you return a Boolean that is equal to `cherrypy.response.status`?

Reply:
3. I've already been thru the mistake of adding such a superfluous success-boolean to an API with later regretting it again. We really shouldn't repeat this mistake.

Reply:
4. Trust the HTTP status code. It's already well established. There is no need to invent something new here.

Reply:
5. If we need to return the name here, then why not return the whole newly generated RBD instead?

Reply:
I'm fine with using HTTP status codes if we can have a different status code for each of the following cases:
* success (e.g., image created)
* validation error (e.g., `image_name` was not provided)
* Ceph library operation exception (e.g., every rbd.OSError)
* unexpected exception (e.g., bug in the code)

But I still have a concern about using HTTP status codes to communicate the result, or error, of a Ceph library operation, when we start executing these operations asynchronously.

I'm preparing a PR where the above RBD create operation is executed asynchronously, and then the result of the operation will be communicated within a wrapper object that describes the async operation.
At the time that the `rbd.create` function returns, it is no longer possible to set the HTTP status code, and the asynchronous execution infrastructure will just return the return value of the async operation, or in case of an exception is raised, it stores the exception object.

Therefore, AFAICT we will not be able to use HTTP status codes for communicating operations results. So do you think it's worth the effort of using HTTP status codes in this PR since we will be changing that in a near future?

Reply:
> If we need to return the name here, then why not instead return the whole newly generated RBD instead?

What do you mean by "newly generated RBD"? Are you suggesting that we perform another rbd library call to fetch the image details and return it to the frontend? or just re-use the parameter values that we received from the frontend?

Reply:
I would suggest the following status codes:

> success (e.g., image created)
-> status 201?
> validation error (e.g., image_name was not provided)
-> status 400? - because it's an error initiated by the user
> Ceph library operation exception (e.g., every rbd.OSError)
-> status 500? - because it's an error caused by the server
> unexpected exception (e.g., bug in the code)
-> status 500? - because it's an error caused by the server as well

@rjfd:
> I'm preparing a PR where the above RBD create operation is executed asynchronously, and
> then the result of the operation will be communicated within a wrapper object that describes the
> async operation.

Does that mean this pull request will become unnecessary anyway?

Reply:
> Does that mean this pull request will become unnecessary anyway?

No, I'm using the code you've written for the RBD creation to test the asynchronous execution infrastructure.

Reply:
Okay :)